charmhub/http.go

```diff
@@ -83,9 +91,19 @@ func (t *APIRequester) Do(req *http.Request) (*http.Response, error) {
 	var potentialInvalidURL bool
 	if resp.StatusCode == http.StatusNotFound {
 		potentialInvalidURL = true
+	} else if resp.StatusCode >= http.StatusInternalServerError && resp.StatusCode <= http.StatusNetworkAuthenticationRequired {
+		defer func() {
+			_, _ = io.Copy(ioutil.Discard, resp.Body)
+			_ = resp.Body.Close()
+		}()
+		return nil, errors.Errorf(`unexpected server error %q`, req.URL.String())
```

Comment:
I don't think we need "unexpected" here. Should we include the status code we got?

Reply:
We dump the response further up, so I don't want to expose internal details.